pkgs/top-level/all-packages.nix

```diff
@@ -309,6 +309,8 @@ with pkgs;
 
   buildMaven = callPackage ../build-support/build-maven.nix {};
 
+  buildIncremental = callPackage ../build-support/build-incremental.nix {};
```

Comment:
I'm not sure about the term "incremental" being used. It's neat that this works, but it's not truly incremental as @roberth already explained. Using "incremental" sets the wrong expectations. Alternatives I can think of:
- Checkpointed builds
- Resumed builds
- Additive builds

Reply:
Ive renamed it to checkpointedBuildTools